tests/testthat/test-recall_data.R

```diff
@@ -0,0 +1,206 @@
+    returned_dtypes <- as.vector(unlist(sapply(result, class)))
+    expect_equal(returned_dtypes, expected_dtypes)
+})
+
+test_that("recall_by_year returns the correct shape",{
+    result <- recall_by_years(2010, 2012)
+    expect_type(result, "list")
+    expect_equal(nrow(result), 25)
+    expected_col_names <- c("Recall number","Manufacturer Name",
+                            "Model name", "Make name", "Year","Recall date")
+    expect_equal(colnames(result), expected_col_names)
+    filtered <- subset(result, Year >= 2010 & Year <= 2012)
+    expect_equal(nrow(result), nrow(filtered))
+    expected_dtypes <- c("character", "character", "character",
+                         "character", "integer",   "Date")
+    returned_dtypes <- as.vector(unlist(sapply(result, class)))
+    expect_equal(returned_dtypes, expected_dtypes)
+})
+
+test_that("recall_by_number returns the correct shape",{
+    result <- recall_by_number(1977044)
+    expect_type(result, "list")
+    expect_equal(nrow(result), 1)
+    expected_col_names <- c("Recall number","Manufacturer Name",
+                            "Model name", "Make name", "Year","Recall date")
+    expect_equal(colnames(result), expected_col_names)
+    expected_dtypes <- c("character", "character", "character",
+                         "character", "integer",   "Date")
+    returned_dtypes <- as.vector(unlist(sapply(result, class)))
+    expect_equal(returned_dtypes, expected_dtypes)
+})
+
+test_that("recall_details returns the correct shape when given an integer",{
+    result <- recall_details(1977044)
+    expect_type(result, "list")
+    expect_equal(nrow(result), 1)
+    expected_col_names <- c("RECALL_NUMBER_NUM","MANUFACTURER_RECALL_NO_TXT", "CATEGORY_ETXT",
+                            "CATEGORY_FTXT", "MODEL_NAME_NM", "MAKE_NAME_NM",
+                            "UNIT_AFFECTED_NBR", "SYSTEM_TYPE_ETXT", "SYSTEM_TYPE_FTXT",
+                            "NOTIFICATION_TYPE_ETXT", "NOTIFICATION_TYPE_FTXT", "COMMENT_ETXT",
+                            "COMMENT_FTXT", "DATE_YEAR_CD", "RECALL_DATE_DTE")
+    expect_equal(colnames(result), expected_col_names)
+    expected_dtypes <- c("character",
+                         "character",
+                         "character",
+                         "character",
+                         "character",
+                         "character",
+                         "integer",
+                         "character",
+                         "character",
+                         "character",
+                         "character",
+                         "character",
+                         "character",
+                         "integer",
+                         "Date")
+    returned_dtypes <- as.vector(unlist(sapply(result, class)))
+    expect_equal(returned_dtypes, expected_dtypes)
+})
+
+test_that("recall_details returns the correct shape when given a list",{
+    recall_numbers <- c(1977044, 2009097)
```

Comment:
if we're testing a list, probably don't need to test the singular

Reply:
done